File: java/src/Arrays/ProductExceptSelf.java
package Arrays;

import java.util.Arrays;

public class ProductExceptSelf {
    public static int[] solution(int[] nums) {
        int[] arr = new int[nums.length];
        int right = 1, left = 1;
//        ITERATE FROM THE LEFT SIDE OF THE ARRAY ALL THE VALUES AND ASSIGN
        for (int i = 0; i < nums.length; i++) {
            arr[i] = left;
            left *= nums[i];
            System.out.println(Arrays.toString(arr));

        }
//        1,1,2,8
        for (int i = nums.length - 1; i >= 0; i--) {
            arr[i] *= right;
            right *= nums[i];
        }
        return arr;
    }
}

File: csharp/Arrays/ProductExceptSelf.cs
namespace csharp.Arrays;

public class ProductExceptSelf
{
    public static int[] Solution(int[] nums)
    {
        int[] answer = new int[nums.Length];
        int counter = 1;
        for (int i = 0; i < nums.Length; i++)
        {
            answer[i] = counter;
            counter *= nums[i];
        }

        counter = 1;
        for (int i = nums.Length - 1; i >= 0; i--)
        {
            answer[i] *= counter;
            counter *= nums[i];
        }

        return answer;
    }
}